### PR TITLE
Rollout management issues refactor

### PR DESCRIPTION
**Pull request by venu1278**
fixes for Rollout management issues.
i)While create new Rollout, RolloutName is not clickable unless status is to change from 'CREATE' to 'READY ' state.
ii)If total number of target groups is zero,then displaying empty.
iii)The style of Action menu is now same as DeploymentView right click option of target table.
iV)The 'Finished %' is dynamically updates in RolloutGroupList while Rollout is started.

Signed-off-by: venugopal boodidadinne venugopal.boodidadinne@in.bosch.com

### CODE REVIEW

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/customrenderers/client/renderers/CustomObjectRenederer.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.ui.customrenderers.client.renderers;
+
+import com.google.gwt.core.shared.GWT;
+import com.vaadin.client.renderers.ClickableRenderer;
+import com.vaadin.client.ui.VButton;
+import com.vaadin.client.widget.grid.RendererCellReference;
+
+public class CustomObjectRenederer extends ClickableRenderer<CustomObject, VButton> {
```
Comment (Jonkno) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/customrenderers/client/renderers/CustomObjectRenederer.java]: Add javadoc

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/customrenderers/renderers/CustomObjectRenderer.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+package org.eclipse.hawkbit.ui.customrenderers.renderers;
+
+import org.eclipse.hawkbit.ui.customrenderers.client.renderers.CustomObject;
+
+import com.vaadin.ui.renderers.ClickableRenderer;
+
+import elemental.json.JsonValue;
+
+public class CustomObjectRenderer extends ClickableRenderer<CustomObject> {
```
Comment (Jonkno) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/customrenderers/renderers/CustomObjectRenderer.java]: Add javadoc

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/customrenderers/renderers/CustomObjectRenderer.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+	/**
+	 *
```
Comment (Jonkno) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/customrenderers/renderers/CustomObjectRenderer.java]: Remove javadoc

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/customrenderers/renderers/CustomObjectRenderer.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	private static final long serialVersionUID = -8754180585906263554L;
+
+	/**
+	 * Creates a new image renderer.
+	 */
+	public CustomObjectRenderer() {
+		super(CustomObject.class, null);
+	}
+
+	public CustomObjectRenderer(Class<CustomObject> presentationType) {
```
Comment (Jonkno) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/customrenderers/renderers/CustomObjectRenderer.java]: Add javadoc

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/ProxyRollout.java [hunk 5]
```diff
@@ -18,113 +19,122 @@
  */
 public class ProxyRollout extends Rollout {
```
Comment (Jonkno) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/ProxyRollout.java]: Add javadoc

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/RolloutListGrid.java [hunk 6]
```diff
@@ -67,505 +68,557 @@
...
+
+	@Autowired
+	private transient RolloutUIState rolloutUIState;
+
+	@Autowired
+	private transient SpPermissionChecker permissionChecker;
+
+	private transient Map<RolloutStatus, StatusFontIcon> statusIconMap = new EnumMap<>(RolloutStatus.class);
+
+	@EventBusListenerMethod(scope = EventScope.SESSION)
```
Comment (Jonkno) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/RolloutListGrid.java]: Add javadoc

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/RolloutListGrid.java [hunk 7]
```diff
@@ -67,505 +68,557 @@
...
+		addTargetWindow.setCaption(i18n.get("caption.update.rollout"));
+		UI.getCurrent().addWindow(addTargetWindow);
+		addTargetWindow.setVisible(Boolean.TRUE);
+	}
+
+	private void refreshGrid() {
+		((LazyQueryContainer) getContainerDataSource()).refresh();
+	}
+
+	public final class FontIconGenerator extends PropertyValueGenerator<String> {
```
Comment (Jonkno) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/RolloutListGrid.java]: add javadoc

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/RolloutListGrid.java [hunk 8]
```diff
@@ -67,505 +68,557 @@
...
+			return String.class;
+		}
+	}
+
+	/**
+	 * 
+	 * Converter to convert 0 to empty , if total target groups is zero.
+	 *
+	 */
+
```
Comment (Jonkno) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/RolloutListGrid.java]: remove empty line and empty javadoc lines

**Comment by Jonkno** [comment]
Hello Venu,
Please refactor your pullrequest regarding missing javadoc etc.
1.	Java doc for all classes und public methods (accept getter and setter)
2.	Remove empty lines in javadoc comments 
`
/** 
* Method description. 
* 
* @param a 
* a …. 
*/

`

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/ProxyRollout.java (hunk 5 above)
Comment (venu1278) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/ProxyRollout.java]: Javadoc is already provided, corrected typo mistake.
Please let me know that expected is same?.

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/customrenderers/client/CustomObjectRendererConnector.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+import com.google.web.bindery.event.shared.HandlerRegistration;
+import com.vaadin.client.connectors.ClickableRendererConnector;
+import com.vaadin.client.renderers.ClickableRenderer.RendererClickHandler;
+import com.vaadin.shared.ui.Connect;
+
+import elemental.json.JsonObject;
+
+@Connect(org.eclipse.hawkbit.ui.customrenderers.renderers.CustomObjectRenderer.class)
+public class CustomObjectRendererConnector extends ClickableRendererConnector<CustomObject> {
```
Comment (michahirsch) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/customrenderers/client/CustomObjectRendererConnector.java]: @venu1278 please provide some javadoc for this class.

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/customrenderers/client/renderers/CustomObject.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.ui.customrenderers.client.renderers;
+
+import java.io.Serializable;
+
+public class CustomObject implements Serializable {
```
Comment (michahirsch) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/customrenderers/client/renderers/CustomObject.java]: @venu1278  please provide some javadoc for this class.

Comment (michahirsch) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/customrenderers/client/renderers/CustomObject.java]: Is there maybe a better class name for `CustomObject` more meaningful name what this Object is doing? holding information? `CustomObject` is for me not meaningful enough :)

**Comment by michahirsch** [comment]
rebuilding circle ci build, because seemed to hang

**Pull request merged**